### PR TITLE
[SPARK-32125][UI] Support get taskList by status in Web UI and SHS Rest API

### PR DESCRIPTION
**Pull request by warrenzhu25**
### What changes were proposed in this pull request?
Support fetching taskList by status as below:
```
/applications/[app-id]/stages/[stage-id]/[stage-attempt-id]/taskList?status=failed
```

### Why are the changes needed?

When there're large number of tasks in one stage, current api is hard to get taskList by status

### Does this PR introduce _any_ user-facing change?
Yes. Updated monitoring doc.

### How was this patch tested?
Added tests in `HistoryServerSuite`

### CODE REVIEW

**Comment by sarutak** [comment]
Hi @warrenzhu25 , thank you for your contribution.
This PR seems to add a new feature so could you add a testcase for it?
You can find tests for the status API in `UISeleniumSuite` and `HistoryServerSuite`.

**Comment by warrenzhu25** [comment]
> Hi @warrenzhu25 , thank you for your contribution.
> This PR seems to add a new feature so could you add a testcase for it?
> You can find tests for the status API in `UISeleniumSuite` and `HistoryServerSuite`.

Added UT, but it seems doc build failed. It seems unrelated with my change.

**Comment by sarutak** [comment]
ok to test.

**Comment by sarutak** [comment]
cc: @squito

**Comment by sarutak** [comment]
retest this please.

File: core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala [hunk 1]
```diff
@@ -96,8 +96,9 @@ private[v1] class StagesResource extends BaseAppResource {
       @PathParam("stageAttemptId") stageAttemptId: Int,
       @DefaultValue("0") @QueryParam("offset") offset: Int,
       @DefaultValue("20") @QueryParam("length") length: Int,
-      @DefaultValue("ID") @QueryParam("sortBy") sortBy: TaskSorting): Seq[TaskData] = {
-    withUI(_.store.taskList(stageId, stageAttemptId, offset, length, sortBy))
+      @DefaultValue("ID") @QueryParam("sortBy") sortBy: TaskSorting,
+      @QueryParam("status") statuses: JList[TaskStatus]): Seq[TaskData] = {
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala:100]: Why do we need to make it as array here? How about just `TaskStatus`?

Comment (warrenzhu25) [reply core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala:100]: Ok, I'm also considering using `TaskStatus`. The reason I use JList is to make it consistent with stageList as below:

```
  @GET
  def stageList(@QueryParam("status") statuses: JList[StageStatus]): Seq[StageData] = {
    withUI(_.store.stageList(statuses))
  }
```

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala]: Oh, sorry I was not aware of this. Please change it back to `JList` to make it consistent.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by sarutak** [comment]
Jenkins will restart 0:00 PDT so we'll need to let Jenkins run again.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
All the dependency failure happens on [jenkins-worker4](https://amplab.cs.berkeley.edu/jenkins/computer/amp-jenkins-worker-04).
It seems worker4 has .m2 corruption issue.

**Comment by sarutak** [comment]
I've reported the .m2 corruption issue (https://issues.apache.org/jira/browse/SPARK-32153#).
So let's try again.
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by warrenzhu25** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by warrenzhu25** [comment]
@gengliangwang Tests passed, could you help merge this?

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**